Common/Config.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System.IO;
+using System.Collections.Generic;
+using System;
+using System.Text.RegularExpressions;
+using System.Linq;
+
+namespace Common
+{
+    /// <summary>
+    /// Stores the uploader's config and applies it depending on the
+    /// </summary>
+    public class Config
+    {
+        /// <summary>
+        /// Path to the config file. It's located one directory above the uploader's and GUI's DLLs.
+        /// </summary>
+        public static readonly string ConfigFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\Profiler.yml");
+
+        /// <summary>
+        /// Thrown if the config for a process is invalid.
+        /// </summary>
+        public class InvalidConfigException : Exception
+        {
+            public InvalidConfigException(string processPath, IEnumerable<string> errors)
+                : base($"Config for process {processPath} is invalid: " + string.Join(", ", errors))
+            {
+            }
+
+            public InvalidConfigException(string message)
+                : base(message)
+            {
+            }
+        }
+
+        /// <summary>
+        /// Config values that apply to a particular process.
+        /// All fields may be null if the user did not explicitly set them
+        /// </summary>
+        public class ConfigForProcess
```

Comment:
ApplySection can't be private since it's used by the outer object. "internal" really doesn't mean anything in a shared project since it'll be compiled into the assembly of the referencing project so it's equivalent to public in our case. Internal really only makes sense for shared assemblies of which we don't have any so I prefer to leave it at public. I'm sure I can trust everyone at CQSE not to do stupid things with the config ;-)

Reply:
Didn't mean it as a vote of no-confidence ;D I usually try to be conservative with visibility, simply because it's easier to make something more public later than to make it more private. You have a point about `internal` being equivalent to `public` though.